### ISSUE TITLE
Local dotnet tool install uses cache even when installing from source dir

### ISSUE THREAD

**Issue by premun** [dotnet/sdk#11484]
I am seeing following behavior:
1. I have an Azure DevOps pipeline
2. In one step, I produce a Debug NuGet package with my tool that I want to test in the next step.
3. The tool has **always the same version** `1.0.0-ci` since it is not being published anywhere.
4. In a next job on a different machine, I download this NuGet as a file and install the tool locally:
```
dotnet new tool-manifest
dotnet tool install --version 1.0.0-ci --add-source [dir] [name of the app]
```
5. The tool installs correctly, but after using it, I find out it is some old cached version installed by older build.

I tried --no-cache but no change, I am definitely sure I use the latest NuGet package because of other files inside.

Is there a way to resolve this other than starting to produce some version numbers?

**Comment by premun** [comment]
FWIW, cleaning the NuGet cache helped me:
```
cache_dirs=`dotnet nuget locals All -l | cut -d':' -f 2 | tr -d ' '`
while IFS= read -r path; do
    rm -rfv "$path/[tool name]"
done <<< "$cache_dirs"
```

**Comment by dsplaisted** [comment]
Yes, NuGet does not make it easy to reuse the same package version for different package contents.
